### ISSUE TITLE
EPIC: Build the entire UX with WinUI Style

### ISSUE THREAD

**Issue by be5invis** [M2Team/NanaZip#5]
As <s>Reunion</s> [WASDK](https://github.com/microsoft/WindowsAppSDK/) is moving close to v1.0, instead of rewriting the UX with XAML Island, using WinUI may be a better idea.

**Comment by fcharlie** [comment]
Good idea.

**Comment by MouriNaruto** [comment]
I have assessed the WinUI 2 and Windows App SDK (Project Reunion) several months ago. Finally I still choose XAML Islands to build the entire UX. Here are some details.

- I will use part of Windows App SDK (MSBuild PriGen targets) and WinUI 2 (Windows 11 control styles and Mica compositor implementation) implementations to achieve the Windows 11 UX style and XAML Islands with single executable. So users still have the modern Windows UX experience.
- Windows App SDK (50MB Runtime) and WinUI 2 (15MB Package) are still too big for the unpackaged version I have planned in the future. The reason for most people who choose the unpackage version is that they don't want to install any packages or runtimes. So I need to include them inside it because these infrastructures are not the builtin components for Windows 10 (I can't find the CBS version in Windows 10 WinSxS folder.). I hope a smaller size footprint.
- The implementation of my UWP XAML adapt layer is called Mile.Xaml. I am developing it internally and will publish them to GitHub publically before I start to develop the NanaZip 2.x series. Mile.Xaml is a lightweight infrastructure and it may provide the static linked library and NanaZip will choose that form to use.

Kenji Mouri

**Comment by be5invis** [comment]
The thing is _WinUI 2 will eventually discontinue and everything moves to WinUI 3_.
Do you think upgrading twice is better than upgrading once?

**Comment by be5invis** [comment]
> So I need to include them inside it because these infrastructures are not the builtin components for Windows 10

They are not built-in for _any_ OS versions AFAIK. Such a decoupling from OS is a design goal of WASDK.

**Comment by MouriNaruto** [comment]
@be5invis 

> *The thing is WinUI 2 will eventually discontinue and everything moves to WinUI 3.*

Upgrade to the XAML control styles and new compositor from WinUI 3 if they do some changes because Microsoft will make WinUI 3 open source. So this project still use UWP XAML.

> They are not built-in for any OS versions AFAIK. Such a decoupling from OS is a design goal of WASDK.

Of course they will built-in unless they don't want to use them to modernize the Windows UX components. I guess the CBS version of WinUI 2/3 will be the fallback when the Windows Store version is not available. (Similar way like the notepad.)

Also, I want to say something about "decoupling from OS". Many years ago, the guys who maintaining the MSVC runtime says the similar words, but the features in builtin `msvcrt.dll` is still update feature sets to MSVC 2008. They finally make the builtin Universal CRT to replace that for using in the new OS components. Today, ``ucrtbase.dll and 'vcruntime140.dll' both have the same part and Windows App SDK uses Universal CRT directly and NanaZip follow the way.

References for WinUI 2 CBS version

- https://github.com/microsoft/microsoft-ui-xaml/blob/c8c76f032c9b44269eb00d5a6f2a54e3e961aca0/dev/dll/XamlControlsResources.cpp#L119
- https://github.com/microsoft/microsoft-ui-xaml/search?q=cbs&type=issues
- ![image](https://user-images.githubusercontent.com/10867563/135399831-2370f5e3-de17-48f1-9240-d5fd001f0f63.png)

Kenji Mouri

**Comment by wangwenx190** [comment]
> 
> 
> > So I need to include them inside it because these infrastructures are not the builtin components for Windows 10
> 
> They are not built-in for _any_ OS versions AFAIK. Such a decoupling from OS is a design goal of WASDK.

@be5invis  I saw the 64 bit runtime dlls of WinUI 3 is around **40MB**, if it's not built in the OS, then every software that based on WinUI 3 has to ship these big dlls? If I develop a small tool which is only 1MB, I still have to ship the 40MB WinUI3 runtime library? That sounds quite ridiculous.

**Comment by be5invis** [comment]
@wangwenx190 From [\[1\]] it looks like you can specify dependencies at MSIX packages so shipping a large DLL is not necessary.

[\[1\]]: https://docs.microsoft.com/en-us/windows/apps/desktop/modernize/framework-packages/framework-packages-overview

**Comment by wjk** [comment]
@be5invis Yes, but then the user will need to have an active Internet connection _and_ the ability to download software from the Microsoft Store in order to install your app. This is one of the main reasons why I don’t use WinUI 3.

**Comment by RGarrido03** [comment]
Hi!

I only knew about this project starting today (through the MDL forum). Congratulations on the progress already done and the plans you made for the future of this app.

Regarding this topic, I made a concept one month ago of what could 7-Zip look like if it used WinUI 2.6. This concept may help you with the redesign you want to do.

Thank you so much for this project.

Rúben Garrido

![image](https://user-images.githubusercontent.com/57329376/135589508-14577cb0-8ab1-4a03-b522-1d1c78b015bd.png)

**Comment by MouriNaruto** [comment]
@RGarrido03 

I have watched your concept in https://sourceforge.net/p/sevenzip/feature-requests/1536 about half a month ago. It inspires me and let me feel make a 7-Zip derivative with modern user experience is meaningful. This is why I create NanaZip.

Thank you so much for your concept.

Kenji Mouri

**Comment by MouriNaruto** [comment]
I think it's time for close the issue because this issue have no further responses.

Kenji Mouri

**Comment by AndromedaMelody** [comment]
- This comment is converted from the discussion (@hyperactiverobot , @MouriNaruto discussed in [#50](https://github.com/M2Team/NanaZip/discussions/50)).
- For more information, please read #274. Thank you.

> ## Will NanaZip design be upgraded to Windows 11?
> <sup>Originally posted by **@hyperactiverobot** November 22, 2021</sup>
> Simply that question, if it will have a modern design or there will be no changes.
>
> > <sup>Originally posted by **@MouriNaruto** November 22, 2021</sup>
> > Starting with NanaZip 2.x Series, NanaZip will be modernized the UI with XAML Islands with the Windows 11 control style, Mica material, dark and light mode support, which is mentioned in https://github.com/M2Team/NanaZip#development-roadmap.
> > 
> > Kenji Mouri
> >
> > > <sup>Originally posted by **@hyperactiverobot** November 22, 2021</sup>
> > > Sorry, I hadn't checked the roadmap. Great news, thanks!

**Comment by RayGicEFL** [comment]
> Finally, this project will upgrade to WinUI 3 and use full feature sets by replacing the namespace if the WinUI 3 are started to be used by Windows OS components. (I mean something in `SystemApps` folder uses it.) And I will change the minimum system requirements to the first Windows version which is really do this.  

A somewhat exciting development.
![Picture](https://user-images.githubusercontent.com/96906415/228137276-1a4bd88c-52e5-4623-bf5d-0fb999103bf9.png)

**Comment by Shomnipotence** [comment]
> > Finally, this project will upgrade to WinUI 3 and use full feature sets by replacing the namespace if the WinUI 3 are started to be used by Windows OS components. (I mean something in `SystemApps` folder uses it.) And I will change the minimum system requirements to the first Windows version which is really do this.
> 
> A somewhat exciting development. ![Picture](https://user-images.githubusercontent.com/96906415/228137276-1a4bd88c-52e5-4623-bf5d-0fb999103bf9.png)

Can I get in touch with you?

**Comment by driver1998** [comment]
I am designing various dialogs in NanaZip. These are early XAML mockups, subject to change.
We are still going to use System XAML Islands via Mile.Xaml for the final UI.

![屏幕截图 2023-07-09 010655](https://github.com/M2Team/NanaZip/assets/22699485/d65dfd02-01c7-412f-81cb-0018301b8127)
![屏幕截图 2023-07-09 010908](https://github.com/M2Team/NanaZip/assets/22699485/41276512-712f-4ba1-a26b-69ac2cc13762)
![屏幕截图 2023-07-09 010931](https://github.com/M2Team/NanaZip/assets/22699485/91e18d3f-522e-403c-8621-f48af26b7f7d)
![屏幕截图 2023-07-09 010942](https://github.com/M2Team/NanaZip/assets/22699485/3bf90acc-b528-4e0d-93c5-07fb71188be3)
